rest/specs/src/main/filtered-resources/openapi/openapi.yaml

```diff
@@ -194,32 +194,19 @@ components:
       description: Transformation request.
       type: object
       required:
-        - id
-        - stagedLocation
-        - productLocation
+        - location
         - mimeType
         - bytes
-        - callbackUrl
       properties:
-        id:
-          type: string
-          description: The unique identifier for a request.
-          minLength: 1
-          maxLength: 255
-          pattern: '^[a-zA-Z0-9]*$'
-          example: 30f14c6c1fc85cba12bfd093aa8f90e3
-        stagedLocation:
-          type: string
-          description: The temporary location of the ingested product.
-          minLength: 1
-          maxLength: 2048
-          example: https://bucket.s3.amazonaws.com
-        productLocation:
+        location:
           type: string
-          description: The final location of the ingested product.
+          format: uri
+          description: >
+            The URI of the ingested product. This URI must support HTTP GET at location/{productId} to retrieve
```

Comment:
❓ Where are `productId` and `metadataType` defined? Does the put only support metadata, or other binary content such as thumbnails or preview clips?

Reply:
My one cent:
Is we call something a "derived product" then we are asserting it is a kind of product. It would be considered a new resource, just like the original product. Like any other product, It could even be fed back into the transform process.
It follows that thumbnails (and metadata) are sub resources and not derived products.

By my reckoning, a good example of a derived product is a NITF that contains 3 different images. Each image could be treated as a product. Another example might be video that contained an extra channel for infrared. The infrared channel could be split out from the RGB channels as a derived product

Reply:
I'm fine with using PUTs to create metadata and thumbnail sub-resources. Tthere are limitations to be aware of. The one that comes to mind is versioning. PUTs replace what exists. That means things like the thumbnail or metadata would be overwritten if updated. I don't think we need to keep a history of  sub-resources but if we did, we would use a POST and also create an ID for sub-resources.